spec/slack/real_time/concurrency/celluloid_spec.rb

```diff
@@ -43,6 +43,22 @@ def read
             socket.run_loop
           end
         end
+
+        describe '#run_loop with epipe' do
+          let(:test_socket) do
```

Comment:
Lets reorganize this a bit? There should be a default test socket for a bunch of tests. Then the `fail` scenarios should be separate with something like 

```ruby
[Errno::EPIPE,  EOFError].each do |ex|
 context "when #{ex} is raised" do
 ...
 end
end
```